crates/modelardb_server/src/storage/uncompressed_data_manager.rs

```diff
@@ -552,14 +565,14 @@ impl UncompressedDataManager {
         Ok(())
     }
 
-    /// Read the next full [`super::uncompressed_data_buffer::UncompressedInMemoryDataBuffer`] from
-    /// a channel, compress it, and send the compressed segments to the
-    /// [`super::CompressedDataManager`] over a channel. Returns [`IOError`] if the error bound
-    /// cannot be retrieved from the [`MetadataManager`].
+    /// Compress `data_buffer` and send the compressed segments to the
+    /// [`CompressedDataManager`]()super::CompressedDataManager over a channel. Returns
+    /// [`SendError`] if a [`Message`] cannot be send to
+    /// [`CompressedDataManager`](super::CompressedDataManager).
     async fn compress_finished_buffer(
         &self,
         mut data_buffer: Box<dyn UncompressedDataBuffer>,
-    ) -> Result<(), IOError> {
+    ) -> Result<(), SendError<Message<CompressedSegmentBatch>>> {
         // unwrap() is safe to use as record_batch() only fail for spilled buffers with no files.
         let data_points = data_buffer.record_batch().await.unwrap();
```

Comment:
After looking around a bit it seems that we do not currently update the used_disk_space_metric when spilled files are deleted (this happens when getting the record batch). If you can find a quick way to do this then maybe just do it in this PR, otherwise maybe just create a bug item on Azure.

Reply:
I have created a bug with a link to this comment.